### PR TITLE
Validate Packages in Zip

### PR DESCRIPTION
**Pull request by anagrius**

### CODE REVIEW

File: api/packages.go [hunk 1]
```diff
@@ -50,10 +50,31 @@ type InstalledPackage struct {
...
+}
+
 // Validate checks a package declaration validity against a Humio
 // server.
 func (p *Packages) Validate(viewName string, absDiretoryPath string) (*ValidationResponse, error) {
-	zipFilePath, err := createTempZipFromFolder(absDiretoryPath)
+	var zipFilePath string
+	var err error
+
+	isDir, err := isDirectory(absDiretoryPath)
```
Comment (danielamkaer) [inline api/packages.go:67]:
```suggestion
	isDir, err := isDirectory(absDirectoryPath)
```

File: api/packages.go [hunk 2]
```diff
@@ -50,10 +50,31 @@ type InstalledPackage struct {
...
+
+	if err != nil {
+		return nil, err
+	}
+
+	if isDir {
+		zipFilePath, err = createTempZipFromFolder(absDiretoryPath)
+	} else {
+		zipFilePath = absDiretoryPath
```
Comment (danielamkaer) [inline api/packages.go:76]: You can test whether the file is definitely a zip by `zip.OpenReader(absDirectoryPath)` and checking for nil error.

File: api/packages.go (hunk 1 above)
Comment (anagrius) [reply api/packages.go:67]: :) You are fast. It is actually a typo everywhere

File: cmd/humioctl/packages_validate.go [hunk 3]
```diff
@@ -30,7 +30,14 @@ func validatePackageCmd() *cobra.Command {
 	cmd := cobra.Command{
 		Use:   "validate [flags] <repo-or-view-name> <package-dir>",
```
Comment (danielamkaer) [inline cmd/humioctl/packages_validate.go:31]:
```suggestion
		Use:   "validate [flags] <repo-or-view-name> <package-dir-or-zip>",
```

**Review by danielamkaer — APPROVED** [review]

File: api/packages.go (hunk 2 above)
Comment (anagrius) [reply api/packages.go:76]: You are VERY welcome to improve it. I don't have the time atm. This is ok for me.

File: cmd/humioctl/packages_validate.go (hunk 3 above)
Comment (anagrius) [reply cmd/humioctl/packages_validate.go:31]: Good point.

**Pull request merged**